website/server/libs/user/index.js

```diff
@@ -102,6 +102,17 @@ function checkPreferencePurchase (user, path, item) {
   return _.get(user.purchased, itemPath);
 }
 
+async function checkNewInputForProfanity (user, res, newValue) {
+  if (stringContainsProfanity(newValue, 'slur') || stringContainsProfanity(newValue)) {
+    if (stringContainsProfanity(newValue, 'slur')) {
```

Comment:
can we avoid calling `stringContainsProfanity()` twice and instead saving the result of the check the first time?